# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/midi/MIDIInputMap.h
# Commit: cfba299e5067
# Full Hash: cfba299e5067a4eab3f3b982374b0770970a3401
# Author: Gabriele Svelto <gsvelto@mozilla.com>
# Date: 2022-04-12 03:57:01
# Regressor Bug: 1758468
# File Overlap Count: 1
# Description:
#   Bug 1758468 - Use stable, anonymous IDs for Web MIDI ports r=padenot
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D142550
# ==============================================================================

diff -r c0eb987d0f76 -r cfba299e5067 dom/midi/MIDIInputMap.h
--- a/dom/midi/MIDIInputMap.h	Mon Apr 11 17:43:08 2022 +0000
+++ b/dom/midi/MIDIInputMap.h	Mon Apr 11 17:43:09 2022 +0000
@@ -7,7 +7,9 @@
 #ifndef mozilla_dom_MIDIInputMap_h
 #define mozilla_dom_MIDIInputMap_h
 
+#include "mozilla/dom/MIDIPort.h"
 #include "nsCOMPtr.h"
+#include "nsTHashMap.h"
 #include "nsWrapperCache.h"
 
 class nsPIDOMWindowInner;
@@ -27,9 +29,15 @@
   explicit MIDIInputMap(nsPIDOMWindowInner* aParent);
   JSObject* WrapObject(JSContext* aCx,
                        JS::Handle<JSObject*> aGivenProto) override;
+  bool Has(nsAString& aId) { return mPorts.Get(aId) != nullptr; }
+  void Insert(nsAString& aId, RefPtr<MIDIPort> aPort) {
+    mPorts.InsertOrUpdate(aId, aPort);
+  }
+  void Remove(nsAString& aId) { mPorts.Remove(aId); }
 
  private:
   ~MIDIInputMap() = default;
+  nsTHashMap<nsString, RefPtr<MIDIPort>> mPorts;
   nsCOMPtr<nsPIDOMWindowInner> mParent;
 };
 